timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -629,9 +668,9 @@ def _gluonts_forecasts_to_data_frame(
 
         item_ids = forecast_index.unique(level=ITEMID)
         if isinstance(forecasts[0], SampleForecast):
-            forecast_df = self._stack_sample_forecasts(forecasts, item_ids)
+            forecast_df = self._stack_sample_forecasts(cast(List[SampleForecast], forecasts), item_ids)
         elif isinstance(forecasts[0], QuantileForecast):
-            forecast_df = self._stack_quantile_forecasts(forecasts, item_ids)
+            forecast_df = self._stack_quantile_forecasts(cast(List[QuantileForecast], forecasts), item_ids)
         elif isinstance(forecasts[0], DistributionForecast):
             forecast_df = self._stack_distribution_forecasts(forecasts, item_ids)
```

Comment:
What if instead of adding assertions for `all(isinstance(...))` in the respective methods, we use these checks as branches for the if conditions?
```python
if all(isinstance(f, SampleForecast) for f in forecasts):
   ....
else:
    raise AssertionError("Expected all forecasts to be of SampleForecast, QuantileForecast or DistributionForecast type")
```
If my intuitive understanding of how type checkers work is correct, this could save us some assertions / `cast` methods down the line.

Reply:
I don't think it's that easy to pull off. For starters, the type checker won't be satisfied with the all(isinstance()) check. Hence skipping this for now.